commit 4e6ffa075843c36c7c9d1a666ea0efaa12a1dc51
Author: Pete Zaitcev <zaitcev@kotori.zaitcev.us>
Date:   Tue Feb 12 18:13:01 2019 -0600

    Remove get_chksum
    
    This popped up because Fedora mandates warning-free builds,
    and get_chksum triggers a warning because it returns an
    unaligned pointer (it is so analigned, static analysis in
    the compiler can detect it).
    
    The easiest fix is to remove it altogether. We think it should
    be safe, because:
    
     - The function is not listed in any headers
     - Its counterpart is called "set_checksum", not "set_chksum"
     - PyECLib does not use it

diff --git a/src/erasurecode.c b/src/erasurecode.c
index dd59ae5..bc401ca 100644
--- a/src/erasurecode.c
+++ b/src/erasurecode.c
@@ -250,7 +250,7 @@ int liberasurecode_backend_available(const ec_backend_id_t backend_id) {
  *
  * @param id - one of the supported backends as
  *        defined by ec_backend_id_t
- * @param ec_args - arguments to the EC backend
+ * @param args - arguments to the EC backend
  *        arguments common to all backends
  *          k - number of data fragments
  *          m - number of parity fragments
@@ -321,7 +321,7 @@ int liberasurecode_instance_create(const ec_backend_id_t id,
 /**
  * Close a liberasurecode instance
  *
- * @param liberasurecode descriptor to close
+ * @param desc - liberasurecode descriptor to close
  */
 int liberasurecode_instance_destroy(int desc)
 {
@@ -529,7 +529,7 @@ int liberasurecode_decode_cleanup(int desc, char *data)
  *
  * @param desc - liberasurecode descriptor/handle
  *        from liberasurecode_instance_create()
- * @param fragments - erasure encoded fragments (> = k)
+ * @param available_fragments - erasure encoded fragments (> = k)
  * @param num_fragments - number of fragments being passed in
  * @param fragment_len - length of each fragment (assume they are the same)
  * @param force_metadata_checks - force fragment metadata checks (default: 0)
diff --git a/src/erasurecode_helpers.c b/src/erasurecode_helpers.c
index 1494587..4be2634 100644
--- a/src/erasurecode_helpers.c
+++ b/src/erasurecode_helpers.c
@@ -178,8 +178,8 @@ uint64_t get_fragment_size(char *buf)
  * Compute a size aligned to the number of data and the underlying wordsize 
  * of the EC algorithm.
  * 
- * @param instance, ec_backend_t instance (to extract args)
- * @param data_len, integer length of data in bytes
+ * @param instance - ec_backend_t instance (to extract args)
+ * @param data_len - integer length of data in bytes
  * @return integer data length aligned with wordsize of EC algorithm
  */
 int get_aligned_data_size(ec_backend_t instance, int data_len)
@@ -487,17 +487,4 @@ inline int set_checksum(ec_checksum_type_t ct, char *buf, int blocksize)
     return 0;
 }
 
-inline uint32_t* get_chksum(char *buf)
-{
-    fragment_header_t* header = (fragment_header_t*) buf;
-
-    assert(NULL != header);
-    if (header->magic != LIBERASURECODE_FRAG_HEADER_MAGIC) {
-        log_error("Invalid fragment header (get chksum)!");
-        return NULL;
-    }
-
-    return (uint32_t *) header->meta.chksum;
-}
-
 /* ==~=*=~==~=*=~==~=*=~==~=*=~==~=*=~==~=*=~==~=*=~==~=*=~==~=*=~==~=*=~== */
